build_tools/hack/build_env_diag.py

```diff
@@ -18,1123 +19,2202 @@
-        )
-        printc(
-            f"""
-    We found your {program} version is {PYTHON_MAJOR_VERSION}.{PYTHON_MINOR_VERSION}.{PYTHON_PATCH_VERSION} <3.8.
-    We also used uv, which uv not support your version.
-    Through {PYTHON_MAJOR_VERSION}.{PYTHON_MINOR_VERSION}.{PYTHON_PATCH_VERSION} version may usable, but we don't gurantee this version is stable for TheRock build and PyTorch.
-    Please use Python 3 version that is supported in bug-fix/security-fix status; We recommends Python 3 version >= 3.9.
+                "err",
+            )
+            _result = None
+
+        elif py.MINOR_VERSION <= 8:
+            _stat = self.msg_stat(
+                "warn", py, f"Found Python {py.version} at {py.exe} {_env_}."
+            )
+            _except = cstring(
+                f"""
+    Found {py.name} version: {py.version}
+    TheRock team still not examine on these older Python versions yet, and seems will be deprecated in future release.
+    The build maybe success, but we do not promise the stability on older versions.
+        traceback: Python3 version may unstable due to Python3 version too old.
+            > expected version: 3.9.X ≤ python ≤ 3.13.X, found {py.version}
     """,
-            Color_Warning,
-        )
+                "warn",
+            )
+            _result = False
 
-    if "VIRTUAL_ENV" in ENV:
-        print(
-            f"{Emoji_Pass} [{program}] Found {program} {PYTHON_MAJOR_VERSION}.{PYTHON_MINOR_VERSION}.{PYTHON_PATCH_VERSION} at: {PYTHON_INSTALL_DIR} (Virtual ENV)"
-        )
-    else:
-        print(
-            f"{Emoji_Warning} [{program}] Found {program} {PYTHON_MAJOR_VERSION}.{PYTHON_MINOR_VERSION}.{PYTHON_PATCH_VERSION} at: {PYTHON_INSTALL_DIR} (Global ENV)"
-        )
-        printc(
-            f"""
-    We found your have {program} program, but it seems like using Global env.
-    Some python packages will include as dependices and might pollute your Global env.
-    If you already generated venv, please activate it:
-        PS > .venv/Scripts/activate.ps1
-        sh $ source .venv/bin/activate
-    If you not generate it, we recommend you use 'uv' to create a virtual env and activate it:
-        PS > uv venv .venv --python <CPython_VERSION>
-        PS > .venv/Scripts/activate.ps1
-
-        sh $ uv venv .venv --python <CPython_VERSION>
-        sh $ source .venv/bin/activate
-
-    traceback: Detected {program} not using Virtual ENV or using Global ENV
+        elif py.MINOR_VERSION >= 14:
+            _stat = self.msg_stat(
+                "warn", py, f"Found Python {py.version} at {py.exe} {_env_}."
+            )
+            _except = cstring(
+                f"""
+    Found {py.name} version: {py.version}
+    TheRock team still not test on these newer Python versions yet.
+    The build maybe success, but we do not promise the stability on new versions.
```

Comment:
I'd simplify this to make it less personal.
```suggestion
    Found {py.name} version: {py.version}
    These newer Python versions are not yet tested.
```

semi-related: https://developers.google.com/style/person